### PR TITLE
feat(service registry): Support artifact references, dynamic artifact types, and artifact create via remote URL on server

### PR DESCRIPTION
**Pull request by jsenko**
Updates Service Registry client version, and adds support for:
- Creating and managing artifacts with references
- Creating a new artifact via URL on server (as opposed to local download which is done currently)
- Changes needed to support dynamic artifact types

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation change
- [ ] Other (please specify)

### CODE REVIEW

**Comment by jsenko** [comment]
Needs the latest Apicurio Registry schema. The schema in https://github.com/redhat-developer/app-services-sdk-core/tree/main/app-services-sdk-go/registryinstance is old and it failed to update https://github.com/redhat-developer/app-services-sdk-core/actions/runs/4293215594/jobs/7480614471 .

**Comment by jsenko** [comment]
Ready for review!

**Comment by dimakis** [comment]
> Ready for review!

hey @jsenko , thanks for the PR. I've had a look over it and it looks good.
I'm just wondering how would I verify this? sorry I don't have a lot of experience with the service registry side of the CLI/ service registry in general

**Comment by jsenko** [comment]
I can ask some people from our team for a review, @SafarMirek did some CLI work before. Alternatively @carlesarnal or @EricWittmann could you try out the new CLI features from a user perspective?

File: pkg/cmd/registry/artifact/util/pretty_print_references.go
Comment (SafarMirek) [inline pkg/cmd/registry/artifact/util/pretty_print_references.go:16]: I would prefer if there was some output when the artifact has not any reference. Now it prints two empty lines which in my opinion is confusing.

File: pkg/cmd/registry/artifact/crud/get/get.go
Comment (SafarMirek) [inline pkg/cmd/registry/artifact/crud/get/get.go:78]: In case there is a references flag, output-file flag is ignored. We can make it exclusive ( cmd.MarkFlagsMutuallyExclusive(..)) or just print the warning, that output file is ignored and output will be on standard output.

File: pkg/cmd/registry/artifact/crud/get/get.go
Comment (SafarMirek) [inline pkg/cmd/registry/artifact/crud/get/get.go:142]: I suggest adding an option to choose the output format like in the other commands.

File: pkg/cmd/registry/artifact/.notes/References.md
Comment (SafarMirek) [inline pkg/cmd/registry/artifact/.notes/References.md:56]: When someone will reproduce this scenario, they will encounter an error, because their artifacts will have different artifact-id. I suggest adding information about that in the example.

+ I would personally love to see the outputs in the example since it would make the example much more clear for me.

File: pkg/cmd/registry/artifact/crud/create/create.go
Comment (SafarMirek) [inline pkg/cmd/registry/artifact/crud/create/create.go:252]: Is there any reason for someone to have a '=' in the reference name or ':' in the groupname or version?

**Review by SafarMirek — CHANGES_REQUESTED** [review]
I have followed the example to reproduce the feature locally and it worked just fine! I have just a few suggestions for the user experience side of the feature.

Good job!

**Comment by SafarMirek** [comment]
LGTM!

**Review by SafarMirek — APPROVED** [review]

**Comment by jsenko** [comment]
@dimakis hi, review done, ready to merge when you have a moment.

**Comment by jsenko** [comment]
> @dimakis hi, review done, ready to merge when you have a moment.

Sorry hold on there is a build failure:/

**Comment by dimakis** [comment]
> 

hey @jsenko, this is a pretty big PR and I don't have the cycles right now to get to it for a review. but I will get to it ASAP 👍

**Comment by jsenko** [comment]
hi @dimakis Can you please approve the workflow run and prioritize a review of this PR? It has been two weeks, and I got a LGTM review from an Apicurio developer as we discussed.

**Comment by dimakis** [comment]
> The Go Build pipeline fails with a docs issue that seems unrelated, anything I should do? I can also rebase the PR to edit the commit message to pass CC validation if needed.

Hey yes, 
running `make generate-docs` will get the docs you need 👍

**Review by dimakis — APPROVED** [review]
LGTM, great work @jsenko 👍

**Comment by jsenko** [comment]
@dimakis Thank you for all the help!

**Pull request merged**
